teuthology/test/test_misc.py

```diff
@@ -26,45 +27,36 @@ def test_sh_truncate(caplog):
 
 def test_sh_fail(caplog):
     with pytest.raises(subprocess.CalledProcessError) as excinfo:
-        misc.sh("/bin/echo -n AB ; /bin/echo C ; exit 111", 2) == "ABC\n"
+        misc.sh("/bin/echo -n AB ; /bin/echo C ; exit 111", 2)
     assert excinfo.value.returncode == 111
     for record in caplog.records:
         if record.levelname == 'ERROR':
             assert ('replay full' in record.message or
                     'ABC\n' == record.message)
 
 def test_sh_progress(caplog):
-    misc.sh("echo AB ; sleep 5 ; /bin/echo C", 2) == "ABC\n"
+    assert misc.sh("echo AB ; sleep 0.1 ; /bin/echo C", 2) == "AB\nC\n"
     records = caplog.records
     assert ':sh: ' in records[0].message
     assert 'AB' == records[1].message
     assert 'C' == records[2].message
-    #
-    # With a sleep 5 between the first and the second message,
-    # there must be at least 2 seconds between the log record
-    # of the first message and the log record of the second one
-    #
-    assert (records[2].created - records[1].created) > 2
+    assert records[2].created > records[1].created
```

Comment:
the comment explaining 5 seconds sleep with at least 2 seconds difference looked confusing for me, but maybe it had some background

Reply:
I agree; I think I wrote it and it still confuses me, so clearly it's of no value!